ui/imports/shared/controls/ContactSelector.qml

```diff
@@ -83,58 +84,100 @@ Item {
-                anchors.left: parent.left
-                anchors.leftMargin: 14
-                anchors.verticalCenter: parent.verticalCenter
                 image.width: (!!selectedContact && !!selectedContact.displayIcon) ? 32 : 0
                 image.height: 32
                 image.source: (!!selectedContact && !!selectedContact.displayIcon) ? selectedContact.displayIcon : ""
                 active: !!selectedContact && !!selectedContact.displayIcon
             }
             StatusBaseText {
                 id: selectedTextField
+                visible: comboBox.control.count > 0
                 text: !!selectedContact ? selectedContact.alias : ""
-                anchors.left: iconImg.right
-                anchors.leftMargin: 4
-                anchors.verticalCenter: parent.verticalCenter
                 font.pixelSize: 15
                 height: 22
                 verticalAlignment: Text.AlignVCenter
                 color: Theme.palette.directColor1
             }
-        }
-        zeroItemsView: Item {
-            height: 186
             StatusBaseText {
-                anchors.fill: parent
+                visible: comboBox.control.count == 0
                 text: qsTr("You don’t have any contacts yet")
-                verticalAlignment: Text.AlignVCenter
-                horizontalAlignment: Text.AlignHCenter
                 font.pixelSize: 13
-                height: 18
                 color: Theme.palette.baseColor1
             }
         }
 
-        selectMenu.delegate: menuItem
-        selectMenu.width: dropdownWidth
+        delegate: StatusItemDelegate {
+            id: itemContainer
+            property var currentContact: Utils.getContactDetailsAsJson(pubKey)
+
+            highlighted: index === comboBox.control.highlightedIndex
+            width: parent.width
+
+            onClicked: {
+                root.selectedContact = itemContainer.currentContact
+            }
+
+            contentItem: RowLayout {
+                spacing: 12
+
+                StatusSmartIdenticon {
+                    image.source: currentContact.displayIcon
+                }
+                ColumnLayout {
+                    Layout.fillWidth: true
+
+                    StatusBaseText {
+                        Layout.fillWidth: true
+                        text: currentContact.alias
+                        font.pixelSize: 15
+                        color: Theme.palette.directColor1
+                    }
+
+                    RowLayout {
+                        Layout.fillWidth: true
+
+                        StatusBaseText {
+                            Layout.fillWidth: true
+                            text: currentContact.name + " • "
+                            visible: currentContact.ensVerified
+                            color: Theme.palette.baseColor1
+                            font.pixelSize: 12
+                        }
+                        StatusBaseText {
```

Comment:
These should really be one text, see:
https://hackmd.io/@status-desktop/B1eOaf-nd/%2FjNzZnH8ZRiynrrXJ--XpmQ#Dynamic

Reply:
Yeah, there's shouldn't be an explicit width limit :) and if yes, `elide` is the way to go